app/controllers/hosts_controller.rb

```diff
@@ -479,6 +479,15 @@ def update_multiple_power_state
     redirect_back_or_to hosts_path
   end
 
+  def show_multiple_power_status
+    find_multiple_ajax
+    res = {}
+    @hosts.includes(:compute_resource).each do |host|
+      res[host.id] = host.supports_power_and_running? rescue false
```

Comment:
I made changes and introduced the "N/A" label for hosts that do not provide info about power state. Now we have:
 ![](http://i.imgur.com/nKjEFJY.png)

I also made changes to render the HTML server side, but this approach has 2 disadvantages:

1) I clutter the controller with methods related to rendering that should not be there. Moving them to concern and including them in controller would be only a small improvement.

2) I just blindly swap page content for whatever server sends me, which means there is no way for me to handle errors. If something goes wrong, hosts controller responds with error message that definitely is not a format I expect. And because I render server-side, I can't even set all labels to "N/A", which means that spinners spin indefinitely - in the best case.

I realize the approach with hiding labels is a workaround, but the labels are where you would expect them in DOM (unlike our modals that we use in the same fashion) and it does not suffer from the points above. But turning this to a React component would probably be the best, I''ll look into it.